http/http-api/src/main/java/com/okta/commons/http/config/HttpClientConfiguration.java

```diff
@@ -37,6 +39,85 @@
+    private static final String MAX_CONNECTIONS_TOTAL_WARNING_MESSAGE = "Bad max connection total value";
+
+    private static final String CONNECTION_VALIDATION_INACTIVITY_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.validateAfterInactivity";
+    private static final int CONNECTION_VALIDATION_INACTIVITY_DEFAULT_VALUE = 2000; // 2sec
+    private static final String CONNECTION_VALIDATION_INACTIVITY_WARNING_MESSAGE = "Invalid max connection inactivity validation value";
+
+    private static final String CONNECTION_TIME_TO_LIVE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.timeToLive";
+    private static final int CONNECTION_TIME_TO_LIVE_DEFAULT_VALUE = 5 * 1000 * 60; // 5 minutes
+    private static final String CONNECTION_TIME_TO_LIVE_WARNING_MESSAGE = "Invalid connection time to live value";
+
+    public String getMaxConnectionsTotalPropertyKey() {
+        return MAX_CONNECTIONS_TOTAL_PROPERTY_KEY;
+    }
+
+    public String getMaxConnectionsPerRoutePropertyKey() {
+        return MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY;
+    }
+
+    public Integer getMaxConnectionPerRouteDefault() {
+        return MAX_CONNECTIONS_PER_ROUTE_DEFAULT_VALUE;
+    }
+
+    public Integer getMaxConnectionTotalDefault() {
+        return MAX_CONNECTIONS_TOTAL_DEFAULT_VALUE;
+    }
+
+    public Integer getMaxConnectionPerRoute() {
+        if(this.maxConnectionPerRoute == null) {
+            this.maxConnectionPerRoute = parseConfigValue(
+                MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY,
+                MAX_CONNECTIONS_PER_ROUTE_DEFAULT_VALUE,
+                MAX_CONNECTIONS_PER_ROUTE_WARNING_MESSAGE);
+        }
+        return this.maxConnectionPerRoute;
+    }
+
+    public Integer getMaxConnectionTotal() {
+        if(this.maxConnectionTotal == null) {
+            this.maxConnectionTotal = parseConfigValue(
+                MAX_CONNECTIONS_TOTAL_PROPERTY_KEY,
+                MAX_CONNECTIONS_TOTAL_DEFAULT_VALUE,
+                MAX_CONNECTIONS_TOTAL_WARNING_MESSAGE);
+        }
+        return this.maxConnectionTotal;
+    }
+
+    public Integer getConnectionValidationInactivity() {
+        if(this.connectionValidationInactivity == null) {
+            this.connectionValidationInactivity = parseConfigValue(
+                CONNECTION_VALIDATION_INACTIVITY_PROPERTY_KEY,
+                CONNECTION_VALIDATION_INACTIVITY_DEFAULT_VALUE,
+                CONNECTION_VALIDATION_INACTIVITY_WARNING_MESSAGE);
+        }
+        return this.connectionValidationInactivity;
+    }
+
+    public int getConnectionTimeToLive() {
+        if(this.connectionTimeToLive == null) {
+            this.connectionTimeToLive = parseConfigValue(
+                CONNECTION_TIME_TO_LIVE_PROPERTY_KEY,
+                CONNECTION_TIME_TO_LIVE_DEFAULT_VALUE,
+                CONNECTION_TIME_TO_LIVE_WARNING_MESSAGE);
+        }
+        return this.connectionTimeToLive;
+    }
```

Comment:
These configurations are specific to HTTP Client.  This needs to be made generic to work with OK HTTP as well
For example, above you have: 
```java
 MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.maxPerRoute"
```

Not all HTTP clients have this type configuration, so these properties should be some sort of generic property map, parsed the same way the baseUrl and other HTTP properties are configured (NOT using system.property).

Using Sys props prevents the properties from being loaded via other property mechanisms supported by the SDK (~/.okta/okta.yaml, environment variables, etc)

Reply:
All logic updated